### PR TITLE
Split PrimaryPreDigest and SecondaryPreDigest

### PR DESCRIPTION
**Pull request by sorpaas**
This splits BABE's `PreDigest` into `PrimaryPreDigest` and `SecondaryPreDigest`, making some code clearer, and also allowing reuse of `SecondaryPreDigest` in Sassafras. Encoding should stay the same.

### CODE REVIEW

**Comment by sorpaas** [comment]
Tested syncing Kusama using this branch for around ~1000 blocks. Not detecting issues.

File: client/consensus/babe/src/lib.rs [hunk 1]
```diff
@@ -62,7 +62,8 @@ pub use sp_consensus_babe::{
 	BabeApi, ConsensusLog, BABE_ENGINE_ID, SlotNumber, BabeConfiguration,
 	AuthorityId, AuthorityPair, AuthoritySignature,
 	BabeAuthorityWeight, VRF_OUTPUT_LENGTH,
-	digests::{PreDigest, CompatibleDigestItem, NextEpochDescriptor},
+	digests::{PreDigest, PrimaryPreDigest, SecondaryPreDigest,
+			  CompatibleDigestItem, NextEpochDescriptor},
```
Comment (andresilva) [inline client/consensus/babe/src/lib.rs]:
```suggestion
	digests::{
		CompatibleDigestItem, NextEpochDescriptor, PreDigest, PrimaryPreDigest, SecondaryPreDigest,
	},
```

File: primitives/consensus/babe/src/digests.rs [hunk 2]
```diff
@@ -27,40 +27,67 @@ use codec::{Decode, Encode};
 #[cfg(feature = "std")]
 use codec::Codec;
 use sp_std::vec::Vec;
+use sp_runtime::RuntimeDebug;
 use sp_consensus_vrf::schnorrkel::{self, Randomness};
 #[cfg(feature = "std")]
 use sp_consensus_vrf::schnorrkel::SignatureError;
 
+/// Raw BABE primary pre-digest.
```
Comment (andresilva) [inline primitives/consensus/babe/src/digests.rs]:
```suggestion
/// Raw BABE primary slot assignment pre-digest.
```

File: primitives/consensus/babe/src/digests.rs [hunk 3]
```diff
@@ -27,40 +27,67 @@ use codec::{Decode, Encode};
...
+	/// Slot number
+	pub slot_number: SlotNumber,
+	/// VRF output
+	pub vrf_output: VRFOutput,
+	/// VRF proof
+	pub vrf_proof: VRFProof,
+}
+
+#[cfg(feature = "std")]
+/// BABE primary pre-digest for std.
```
Comment (andresilva) [inline primitives/consensus/babe/src/digests.rs]:
```suggestion
/// BABE primary slot assignment pre-digest for std environment.
```

File: primitives/consensus/babe/src/digests.rs [hunk 4]
```diff
@@ -27,40 +27,67 @@ use codec::{Decode, Encode};
...
+		Ok(PrimaryPreDigest {
+			authority_index: raw.authority_index,
+			slot_number: raw.slot_number,
+			vrf_output: raw.vrf_output.try_into()?,
+			vrf_proof: raw.vrf_proof.try_into()?,
+		})
+	}
+}
+
+/// BABE secondary pre-digest.
```
Comment (andresilva) [inline primitives/consensus/babe/src/digests.rs]:
```suggestion
/// BABE secondary slot assignment pre-digest.
```

**Pull request merged**